### ISSUE TITLE
can't build standalone mkFit from current CMSSW release

### ISSUE THREAD

**Issue by srlantz** [trackreco/mkFit-external#5]
Compilation with gcc-10 is currently unsuccessful when following the previously successful instructions in https://github.com/cms-sw/cmssw/blob/master/RecoTracker/MkFitCore/standalone/README_buildFromCMSSW.txt.

The failures occur in RecoTracker/MkFitCMS/standalone/Geoms/CMS-2017.cc, which is `#include`'d from mkFit-external/mkfit-geom-cms-2017/CMS-2017.cc. The errors are related to Track variables declared in this file, and they look like this:

```
In file included from /home/users/slantz/mkFitFromCMSSW/RecoTracker/MkFitCMS/standalone/Geoms/CMS-2017.cc:7:
../mkFit-external/mkfit-geom-cms-2017/CMS-2017.cc: In function 'void {anonymous}::partitionSeeds0(const mkfit::TrackerInfo&, const TrackVec&, const mkfit::EventOfHits&, mkfit::IterationSeedPartition&)':
../mkFit-external/mkfit-geom-cms-2017/CMS-2017.cc:344:30: error: invalid use of incomplete type 'const class mkfit::Track'
  344 |       const bool z_dir_pos = S.pz() > 0;
      |                              ^
In file included from ../mkFit-external/mkfit-geom-cms-2017/CMS-2017.cc:8,
                 from /home/users/slantz/mkFitFromCMSSW/RecoTracker/MkFitCMS/standalone/Geoms/CMS-2017.cc:7:
/home/users/slantz/mkFitFromCMSSW/RecoTracker/MkFitCore/interface/IterationConfig.h:14:9: note: forward declaration of 'class mkfit::Track'
   14 |   class Track;
      |         ^~~~~
In file included from /home/users/slantz/mkFitFromCMSSW/RecoTracker/MkFitCMS/standalone/Geoms/CMS-2017.cc:7:
../mkFit-external/mkfit-geom-cms-2017/CMS-2017.cc:346:24: error: invalid use of incomplete type 'const class mkfit::Track'
  346 |       HitOnTrack hot = S.getLastHitOnTrack();
      |                        ^
In file included from ../mkFit-external/mkfit-geom-cms-2017/CMS-2017.cc:8,
                 from /home/users/slantz/mkFitFromCMSSW/RecoTracker/MkFitCMS/standalone/Geoms/CMS-2017.cc:7:
/home/users/slantz/mkFitFromCMSSW/RecoTracker/MkFitCore/interface/IterationConfig.h:14:9: note: forward declaration of 'class mkfit::Track'
   14 |   class Track;
      |         ^~~~~
...
```
The error is avoided, and the build succeeds, if branch CMSSW_12_4_0_pre3 is instead checked out from CMSSW and mkFit-external is backed off to commit `c772192`. It is not at all clear what changes in mkFit (or elsewhere) in the latest CMSSW are leading to the above errors.

**Comment by slava77** [comment]
the standard practice is to include the header if the type data is used in the current file.
Does it compile after including the Track.h in CMS-2017.cc?

**Comment by srlantz** [comment]
> the standard practice is to include the header if the type data is used in the current file.
> Does it compile after including the Track.h in CMS-2017.cc?

That helps. The error now switches to 3 occurrences of this:

```
In file included from /home/users/slantz/mkFitFromCMSSW/RecoTracker/MkFitCMS/standalone/Geoms/CMS-2017.cc:7:
../mkFit-external/mkfit-geom-cms-2017/CMS-2017.cc: In function 'void {anonymous}::partitionSeeds0(const mkfit::TrackerInfo&, const TrackVec&, const mkfit::EventOfHits&, mkfit::IterationSeedPartition&)':
../mkFit-external/mkfit-geom-cms-2017/CMS-2017.cc:416:12: error: 'class mkfit::IterationSeedPartition' has no member named 'm_sort_score'
  416 |       part.m_sort_score[i] = 7.0f * (reg - 2) + eta;
      |            ^~~~~~~~~~~~
```

**Comment by srlantz** [comment]
Closing, I verified that this was fixed by Matevz's commit 901dd41.
